applications/fishing-map/src/features/timebar/TimebarSettings.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { Fragment, useState } from 'react'
+import cx from 'classnames'
+import { useSelector } from 'react-redux'
+import useClickedOutside from 'hooks/useClickedOutside'
+import { TimebarEvents, TimebarGraphs } from 'types'
+import { IconButton, Radio, Select, SelectOption } from '@globalfishingwatch/ui-components/dist'
+import {
+  selectActiveTemporalgridDataviews,
+  selectActiveVesselsDataviews,
+} from 'features/workspace/workspace.selectors'
+import { timebarEventsOptions, timebarGraphOptions } from 'data/config'
+import {
+  selectTimebarEvents,
+  selectTimebarGraph,
+  selectTimebarVisualisation,
+} from 'routes/routes.selectors'
+import { useLocationConnect } from 'routes/routes.hook'
+import styles from './TimebarSettings.module.css'
+
+const TimebarSettings = () => {
+  const [optionsPanelOpen, setOptionsPanelOpen] = useState(false)
+  const activeHeatmapDataviews = useSelector(selectActiveTemporalgridDataviews)
+  const activeVesselDataviews = useSelector(selectActiveVesselsDataviews)
+  const timebarVisualisation = useSelector(selectTimebarVisualisation)
+  const timebarEvents = useSelector(selectTimebarEvents)
+  const timebarGraph = useSelector(selectTimebarGraph)
+  const { dispatchQueryParams } = useLocationConnect()
+
+  const openOptions = () => {
+    setOptionsPanelOpen(true)
+  }
+  const closeOptions = () => {
+    setOptionsPanelOpen(false)
+  }
+  const setHeatmapActive = () => {
+    dispatchQueryParams({ timebarVisualisation: 'heatmap' })
+  }
+  const setVesselActive = () => {
+    dispatchQueryParams({ timebarVisualisation: 'vessel' })
+  }
+  const setEventsOption = (o: SelectOption) => {
+    dispatchQueryParams({ timebarEvents: o.id as TimebarEvents })
+  }
+  const setGraphOption = (o: SelectOption) => {
+    dispatchQueryParams({ timebarGraph: o.id as TimebarGraphs })
+  }
+  const removeEventsOption = () => {
+    dispatchQueryParams({ timebarEvents: 'none' })
+  }
+  const removeGraphOption = () => {
+    dispatchQueryParams({ timebarGraph: 'none' })
+  }
+  const expandedContainerRef = useClickedOutside(closeOptions)
+
+  if (activeHeatmapDataviews?.length === 0 && timebarVisualisation === 'heatmap')
+    dispatchQueryParams({ timebarVisualisation: 'vessel' })
```

Comment:
There are problems with this:
- it might spam the whole app by changing the query params whenever this component is rendered
(having imperative code that affects the whole app state in a component render is not a good idea)
- you are not actually keeping the user preference when defaulting to vessel

See another approach here https://github.com/GlobalFishingWatch/frontend/pull/152/commits/601c3d2729afb0a690e94085e26e83f7b191ac7d
(you could also do that in a selector)